pkg/operator/ceph/csi/spec.go

```diff
@@ -556,17 +556,32 @@ func createCSIDriverInfo(clientset kubernetes.Interface, name string, ownerRef *
 		},
 	}
 	csidrivers := clientset.StorageV1beta1().CSIDrivers()
-	k8sutil.SetOwnerRef(&csiDriver.ObjectMeta, ownerRef)
-	_, err := csidrivers.Create(csiDriver)
+	driver, err := csidrivers.Get(name, metav1.GetOptions{})
 	if err == nil {
-		logger.Infof("CSIDriver object created for driver %q", name)
-		return nil
-	}
-	if apierrors.IsAlreadyExists(err) {
+		if len(driver.ObjectMeta.OwnerReferences) > 0 {
+			// remove the owner reference set by operator on the csidriver
+			for i, o := range driver.ObjectMeta.OwnerReferences {
+				if o.Name == ownerRef.Name {
+					driver.ObjectMeta.OwnerReferences = append(driver.ObjectMeta.OwnerReferences[:i], driver.ObjectMeta.OwnerReferences[i+1:]...)
```

Comment:
What owner refs are being appended? I thought we just don't want owner refs anymore for this.

Reply:
This was done as part of the upgrade. just want to make sure we don't remove any other auto set owner reference (assigning nil and append remains the same as we will have a single reference for now)